compendium/DeclarativeServices/test/TestConfigPolicies.cpp

```diff
@@ -25,110 +25,53 @@
-        }
-      },
-      [&compConfigs, &instance, &uniqueProp, &i]() -> bool {
-        if (compConfigs.size() == 1) {
-          auto properties = compConfigs.at(0).properties;
-          auto id = properties.find(uniqueProp[i]);
-          if (id != properties.end()) {
-            return (id->second == instance[i]);
-          }
-        }
-        return false;
-      });
-
-      ASSERT_TRUE(result) << "Timed out waiting for Update Configuration"
-                             "to complete.";
-      EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
-      EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
-        << "Component instance state should be ACTIVE";
-
-      //Request a service reference to the new component instance. This will
-      //cause DS to construct the instance with the updated properties.
-      auto instanceI = GetInstance<test::CAInterface>();
-      ASSERT_TRUE(instanceI) << "GetService failed for CAInterface";
-
-      //Confirm component instance was created with the correct properties
-      instanceProps = instanceI->GetProperties();
-      auto uniquePropI = instanceProps.find(uniqueProp[i]);
-
-      EXPECT_TRUE(uniquePropI != instanceProps.end())
-        << uniqueProp[i] << " not found in constructed instance";  //not found
-      EXPECT_EQ(uniquePropI->second, instance[i]);
-
-    ++i;
-  }
-}
-
-TEST_F(tServiceComponent, testOptionalConfigPolicyImmediateFalse) // DS_CAI_FTC_5
+  // Confirm component instance was created with the properties has default value
+  auto instanceProps = instance->GetProperties();
+  EXPECT_TRUE(instanceProps.empty())
+    << "Property instance should be empty";
+} // end of testOptionalConfigPolicyWithoutConfigObj
+
+/**
+* Test the component instance is constructed with the configuration object is present
+* and the component should be constructed with the correct properties.
+*/
+TEST_F(tServiceComponent, testOptionalConfigPolicyWithConfigObj) // DS_CAI_FTC_5
 {
-
   // Start the test bundle containing the component name.
-  std::string componentName           = "sample::ServiceComponentCA1";
-  cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSCA1");
+  std::string componentName           = "sample::ServiceComponentCA05a";
+  cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSCA05a");
 
   // Use DS runtime service to validate the component description and
   // Verify that DS is finished creating the component data structures.
   scr::dto::ComponentDescriptionDTO compDescDTO;
   auto compConfigs =
     GetComponentConfigs(testBundle, componentName, compDescDTO);
   EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
-  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED)
-    << "Component state should be SATISFIED";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::ACTIVE)
+    << "Component state should be ACTIVE";
```

Comment:
I think what we want to show in this test is that the configuration properties are used when the component instance is constructed so the configuration object has to be created before this point. One way is to use immediate = false, start the bundle, create the configuration object and then get an instance. The second way to do it is to create the configuration object before you start the bundle.